### ISSUE TITLE
networkd unit tests break connectivity on host

### ISSUE THREAD

**Issue by bluca** [systemd/systemd#20862]
While debugging https://github.com/systemd/systemd/pull/20837 @evverx realized some of the tests break local connectivity on the host they are running in:

https://github.com/coverallsapp/github-action/issues/96#issuecomment-927009647

The CI issue was solved by running the unit tests in a network namespace.

@evverx you managed to reproduce the problem locally, right?

**Comment by bluca** [comment]
The symptom is that after running the whole test suite with:

```
meson --werror -Dtests=unsafe -Dslow-tests=true -Dfuzz-tests=true build
meson test -C build --print-errorlogs
```

coveralls.io could no longer be resolved

**Comment by evverx** [comment]
@bluca I can't seem to reproduce it on my machine. I think maybe it has something to do with how VMs on GHActions are configured.

**Comment by evverx** [comment]
I've just spun up an Ubuntu Focal VM to more or less replicate the GHActions environment and it appears there it can be reproduced reliable. The culprit is `test-network` as far as I can tell
```
vagrant@ubuntu-focal:~/systemd$ resolvectl query ya.ru
ya.ru: 87.250.250.242                          -- link: enp0s3

-- Information acquired via protocol DNS in 3.0ms.
-- Data is authenticated: no

vagrant@ubuntu-focal:~/systemd$ sudo ./build/test-network
filename:1: Specified route table name and number pair conflicts with others, ignoring assignment: ddd:22222
/usr/lib/systemd/network/80-container-ve.network:23: IPMasquerade=yes is deprecated, and it is handled as "ipv4" instead of "both". Please use "ipv4" or "both".
/usr/lib/systemd/network/80-container-vz.network:22: IPMasquerade=yes is deprecated, and it is handled as "ipv4" instead of "both". Please use "ipv4" or "both".
enp0s3: Link UP
enp0s3: Gained carrier
lo: Link UP
lo: Gained carrier
enp0s3: Gained IPv6LL

vagrant@ubuntu-focal:~/systemd$ resolvectl query ya.ru
ya.ru: resolve call failed: No appropriate name servers or networks for name found
```
Apparently it can be fixed by simply restarting `systemd-networkd`.

**Comment by yuwata** [comment]
Fix is waiting in #20877.
